core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -792,6 +793,11 @@ private[spark] object Utils extends Logging {
   lazy val localIpAddress: String = findLocalIpAddress()
   lazy val localIpAddressHostname: String = getAddressHostName(localIpAddress)
 
+  def localIpURI(): String = {
```

Comment:
I don't think it's worth it. Right now the PR is changing behavior, which is a lot worse. I think this change can be concluded quite simply, right? In the 1-2 places where the host address / IP needs to be escaped to be part of a URI, use the `InetAddresses` utility to do so. Adding a utility method to call a utility method doesn't seem like a win.